activemodel/lib/active_model/attribute_set/builder.rb

```diff
@@ -76,15 +76,23 @@ def ==(other)
     end
 
     def marshal_dump
-      materialize
+      if @materialized
+        materialize
+      else
+        [@types, @values, @additional_types, @default_attributes]
```

Comment:
This will lose any mutations that have been made to the record. Can we just stick `@delegate_hash` in here and lose the materialized branch entirely?